packages/components/input/src/input.vue

```diff
@@ -443,6 +461,12 @@ export default defineComponent({
       emit('keydown', evt)
     }
 
+    const handlePaste = (evt: ClipboardEvent) => {
+      const preText = nativeInputValue.value
+      if (toArray(preText).length >= Number(attrs.value.maxlength)) return
+      setHtmlMaxLength(preText + (evt.clipboardData?.getData('text') ?? ''))
+    }
```

Comment:
**Praise:** this is really something we needed!

Reply:
BTW, do we need to determine the type of `attrs.value.maxlength` before converting it to `number`?

Reply:
I'm for it. I have already fixed.

Reply:
I don't think so. Because attrs.value.maxlength maybe string or number. If I got NaN from Number(attrs.value.maxlength)), setHtmlMaxLength can handle this,

Reply:
Sure! Mark as resolved